### ISSUE TITLE
Keep getting this error but I don't see any error: E0001:invalid syntax (<string>, line 631)

### ISSUE THREAD

**Issue by xiaoyanqu** [PyCQA/pylint#2495]
### Steps to reproduce
1. No need to re-produce, it's still there

### Current behavior
This statement (line #631) was caught by pylint but I don't see any issue:
```for (k,v) in sorted(report.items(), key=lambda (k,v): v['host_count']):```

### Expected behavior
No error

### pylint --version output
pylint --version
No config file found, using default configuration
pylint 1.8.4,
astroid 1.6.3
Python 3.6.5 |Anaconda, Inc.| (default, Apr 26 2018, 08:42:37)
[GCC 4.2.1 Compatible Clang 4.0.1 (tags/RELEASE_401/final)]

**Comment by xiaoyanqu** [comment]
VS Code Version 1.27.2 (1.27.2)

**Comment by xiaoyanqu** [comment]
MacBook Pro MacOS Version: 10.13.6

**Comment by vorbit** [comment]
@TheSnoozer is right, that's the exact problem pylint is complaining here. Since we also use the builtin Python parser to parse the source code into ASTs, we're limited to what that module can provide us in terms of errors, which is why you are seeing that `SyntaxError` as the Python file is not parseable due to that.

**Comment by PCManticore** [comment]
@TheSnoozer is right, that's the exact problem pylint is complaining here. Since we also use the builtin Python parser to parse the source code into ASTs, we're limited to what that module can provide us in terms of errors, which is why you are seeing that SyntaxError as the Python file is not parseable due to that.

**Comment by xiaoyanqu** [comment]
Thank you @TheSnoozer and @PCManticore for your prompt reply!

Hmmm.. it works in my python 2.x scripts though. Those links are very helpful, thanks a lot! I'll follow the documentations since now.
